File: Day 20 DSA (1979)/Code.cpp
class Solution(object):
    def findGCD(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        smallest = min(nums)
        largest = max(nums)
        def gcd(a, b):
            while b:
                a, b = b, a % b
            return a
        
        return gcd(smallest, largest)

File: Day 33 DSA (2697)/Code.py
class Solution(object):
    def makeSmallestPalindrome(self, s):
        """
        :type s: str
        :rtype: str
        """
        letters = list(s)

        for i in range(len(s) // 2):
            letters[i] = letters[~i] = min(letters[i], letters[~i])

        return ''.join(letters)
        
